room_assistance/indico_room_assistance/forms.py

```diff
@@ -0,0 +1,62 @@
+# This file is part of the CERN Indico plugins.
+# Copyright (C) 2014 - 2018 CERN
+#
+# The CERN Indico plugins are free software; you can redistribute
+# them and/or modify them under the terms of the MIT License; see
+# the LICENSE file for more details.
+
+from __future__ import unicode_literals
+
+from datetime import date, datetime, time, timedelta
+
+from flask import request, session
+from wtforms import SelectField
+from wtforms.fields.html5 import IntegerField
+from wtforms.fields.simple import TextAreaField
+from wtforms.validators import DataRequired, NumberRange, Optional, ValidationError
+
+from indico.modules.events.requests import RequestFormBase
+from indico.modules.rb import Location
+from indico.util.date_time import now_utc
+from indico.web.forms.base import IndicoForm, generated_data
+from indico.web.forms.fields import IndicoDateField, IndicoDateTimeField
+from indico.web.forms.validators import DateTimeRange, Exclusive
+
+from indico_room_assistance import _
+
+
+class RoomAssistanceRequestForm(RequestFormBase):
+    start_dt = IndicoDateTimeField(_('When'), [DataRequired(),
+                                               DateTimeRange(earliest=datetime.combine(now_utc().date(), time()))],
+                                   description=_('When do you need the assistance?'))
+    reason = TextAreaField(_('Reason'), [DataRequired()], description=_('Why are you requesting assistance?'))
+
+    def validate_start_dt(self, field):
+        localized_time = field.data.astimezone(session.tzinfo).time()
+        is_in_working_hours = any(period[0] <= localized_time <= period[1] for period in Location.working_time_periods)
```

Comment:
needs to be updated. feel free to copy the working time periods into this plugin. having "working time periods" hardcoded in the core isn't amazing anyway, so if we ever make them more dynamic for the occupancy calculation there we don't have to worry about updating the plugin as well.